Possible data race during write of size 1 at 0x4A82ED0 by thread #2
Locks held: 1, at address 0x4A820D0
   at 0x48488CC: mempcpy (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
   by 0x491C661: _IO_new_file_xsputn (fileops.c:1236)
   by 0x491C661: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
   by 0x49047DC: __vfprintf_internal (vfprintf-internal.c:1687)
   by 0x48EED6E: printf (printf.c:33)
   by 0x4015DD: putstatus (routine.c:17)
   by 0x401628: tryeat (routine.c:24)
   by 0x401809: routine (routine.c:71)
   by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
   by 0x4872608: start_thread (pthread_create.c:477)
   by 0x49AC162: clone (clone.S:95)
 Address 0x4a82ed0 is 0 bytes inside a block of size 1,024 alloc'd
   at 0x483C893: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
   by 0x490ED33: _IO_file_doallocate (filedoalloc.c:101)
   by 0x491EEFF: _IO_doallocbuf (genops.c:347)
   by 0x491DF5F: _IO_file_overflow@@GLIBC_2.2.5 (fileops.c:745)
   by 0x491C6E4: _IO_new_file_xsputn (fileops.c:1244)
   by 0x491C6E4: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
   by 0x49047DC: __vfprintf_internal (vfprintf-internal.c:1687)
   by 0x48EED6E: printf (printf.c:33)
   by 0x4015DD: putstatus (routine.c:17)
   by 0x401628: tryeat (routine.c:24)
   by 0x401809: routine (routine.c:71)
   by 0x4842B1A: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
   by 0x4872608: start_thread (pthread_create.c:477)
 Block was alloc'd by thread #3

---Thread-Announcement------------------------------------------

Thread #1 is the program's root thread

----------------------------------------------------------------

Thread #1: pthread_mutex_destroy of a locked mutex
   at 0x483FC96: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
   by 0x401554: end_check (routine_utils.c:58)
   by 0x401CFB: sit_philo (main.c:83)
   by 0x401E55: set_table (main.c:113)
   by 0x401EB9: main (main.c:123)

----------------------------------------------------------------

Thread #1's call to pthread_mutex_destroy failed
   with error code 16 (EBUSY: Device or resource busy)
   at 0x483FD60: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
   by 0x401554: end_check (routine_utils.c:58)
   by 0x401CFB: sit_philo (main.c:83)
   by 0x401E55: set_table (main.c:113)
   by 0x401EB9: main (main.c:123)

----------------------------------------------------------------

Thread #1: pthread_mutex_destroy of a locked mutex
   at 0x483FC96: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
   by 0x401572: end_check (routine_utils.c:59)
   by 0x401CFB: sit_philo (main.c:83)
   by 0x401E55: set_table (main.c:113)
   by 0x401EB9: main (main.c:123)

----------------------------------------------------------------

Thread #1's call to pthread_mutex_destroy failed
   with error code 16 (EBUSY: Device or resource busy)
   at 0x483FD60: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
   by 0x401572: end_check (routine_utils.c:59)
   by 0x401CFB: sit_philo (main.c:83)
   by 0x401E55: set_table (main.c:113)
   by 0x401EB9: main (main.c:123)

----------------------------------------------------------------

Thread #1: pthread_mutex_destroy of a locked mutex
   at 0x483FC96: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
   by 0x40158A: end_check (routine_utils.c:60)
   by 0x401CFB: sit_philo (main.c:83)
   by 0x401E55: set_table (main.c:113)
   by 0x401EB9: main (main.c:123)

----------------------------------------------------------------

Thread #1's call to pthread_mutex_destroy failed
   with error code 16 (EBUSY: Device or resource busy)
   at 0x483FD60: ??? (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_helgrind-amd64-linux.so)
   by 0x40158A: end_check (routine_utils.c:60)
   by 0x401CFB: sit_philo (main.c:83)
   by 0x401E55: set_table (main.c:113)
   by 0x401EB9: main (main.c:123)

----------------------------------------------------------------

Thread #1: Exiting thread still holds 1 lock
   at 0x4970176: _Exit (_exit.c:31)
   by 0x48D39F1: __run_exit_handlers (exit.c:132)
   by 0x48D3A8F: exit (exit.c:139)
   by 0x48B10B9: (below main) (libc-start.c:342)


Use --history-level=approx or =none to gain increased speed, at
the cost of reduced accuracy of conflicting-access information
For lists of detected and suppressed errors, rerun with: -s
ERROR SUMMARY: 987 errors from 10 contexts (suppressed: 902 from 104)
